java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -222,17 +259,38 @@ private static void allocateVectors(VectorSchemaRoot root, int size) {
    * Iterate the given JDBC {@link ResultSet} object to fetch the data and transpose it to populate
    * the given Arrow Vector objects.
    *
-   * @param rs   ResultSet to use to fetch the data from underlying database
-   * @param root Arrow {@link VectorSchemaRoot} object to populate
+   * @param rs       ResultSet to use to fetch the data from underlying database
+   * @param root     Arrow {@link VectorSchemaRoot} object to populate
+   * @param calendar The calendar to use when reading time-based data.
    * @throws SQLException on error
    */
   public static void jdbcToArrowVectors(ResultSet rs, VectorSchemaRoot root, Calendar calendar)
       throws SQLException, IOException {
 
     Preconditions.checkNotNull(rs, "JDBC ResultSet object can't be null");
-    Preconditions.checkNotNull(root, "JDBC ResultSet object can't be null");
+    Preconditions.checkNotNull(root, "Vector Schema cannot be null");
     Preconditions.checkNotNull(calendar, "Calendar object can't be null");
 
+    jdbcToArrowVectors(rs, root, new JdbcToArrowConfig(new RootAllocator(0), calendar));
```

Comment:
I think, RootAllocator with argument 0, may not behave as expected. Have you tested this?

Reply:
This is a little confusing.  The config's allocator is used in the `JdbcToArrow` static member functions to construct a `VectorSchemaRoot`.  At this point, the `VectorSchemaRoot` has already been constructed, so the allocator is an unused argument.  That's why its allocation can be `0`.

Reply:
Yea, got it. Mainly  for the backeward compatibilit for JdbcToArrowUtils.java class.